File: RatedLists.Frontend/src/app/models/listOfItems.ts
import {Item, ITEMS} from './item';

export class ListOfItems {
 name: string;
 items: Item[];
}
export const LISTOFITEMS: ListOfItems[] = [
  {
    name: 'Name11',
    items: [ITEMS[0]]
  },
  {
    name: 'Name22',
    items: [ITEMS[1], ITEMS[0]]
  },
  {
    name: 'Name33',
    items: [ITEMS[2], ITEMS[1]]
  },
];

File: RatedLists.Backend/RatedLists.Backend/Models/ListOfItems.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;

namespace RatedLists.Backend.Models
{
    public class ListOfItems
    {
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string Name { get; set; }

        public List<Item> Items { get; set; }
    }
}
